examples/solvers/functional_basic_example.py

```diff
@@ -0,0 +1,252 @@
+else:
+    print('There should have been an error, but it did not occure.')
+
+# We now create the conjugate functional and test a call to it
+my_func_conj = my_func.conjugate_functional
+
+if my_func_conj(x) == (x.norm()**2 / 2.0 - (x - my_func.y).norm()**2 / 4.0 +
+                       my_func.y.norm()**2 / 2.0 - x.inner(my_func.y)):
+    print('The conjugate functional evaluates correctly.')
+else:
+    print('There is an error in the evaluation of the conjugate functional.')
+
+# Nothing else has been implemented in the conjugate functional. For example,
+# there is no gradient.
+try:
+    my_func_conj.gradient
+except NotImplementedError:
+    print('As expected we caught a NotImplementedError when trying to access '
+          'the gradient operator.')
+else:
+    print('There should have been an error, but it did not occure.')
+
+# There is no derivative either.
+try:
+    my_func_conj.derivative(x)(p)
+except NotImplementedError:
+    print('As expected we caught a NotImplementedError when trying to '
+          'evaluate the derivative.')
+else:
+    print('There should have been an error, but it did not occure.')
+
+# We now test some general properties that exists for all functionals. We can
+# add two functioanls, scale it by multiplying with a scalar from the left,
+# scale the argument by multiplying with a scalar from the right, and also
+# translate the argument. Except for the sum of two functional, the other
+# operations will apply corrections in order to evaluate gradients, etc.,
+# in a correct way.
+
+# Scaling the functional
+func_scal = np.random.rand()
+my_func_scaled = func_scal * my_func
+
+if my_func_scaled(x) == func_scal * (my_func(x)):
+    print('Scaling of functional works.')
+else:
+    print('There is an error in the scaling of functionals.')
+
+my_func_scaled_grad = my_func_scaled.gradient
+if my_func_scaled_grad(x) == func_scal * (my_func.gradient(x)):
+    print('Scaling of functional evaluates gradient correctly.')
+else:
+    print('There is an error in evaluating the gradient in the scaling of '
+          'functionals.')
+
+# Scaling of the argument
+arg_scal = np.random.rand()
+my_func_arg_scaled = my_func * arg_scal
+
+if my_func_arg_scaled(x) == my_func(arg_scal * x):
+    print('Scaling of functional argument works.')
+else:
+    print('There is an error in the scaling of functional argument.')
+
+# Sum of two functionals
+y_2 = space.element(np.random.randn(n))
+my_func_2 = MyFunctional(domain=space, y=y_2)
+my_funcs_sum = my_func + my_func_2
+
+if my_funcs_sum(x) == my_func(x) + my_func_2(x):
+    print('Summing two functionals works.')
+else:
+    print('There is an error in the summation of functionals.')
+
+# Translation of the functional, i.e., creating the functional f(. - y).
+my_func_translated = my_func.translate(y_2)
+if my_func_translated(x) == my_func(x - y_2):
+    print('Translation of functional works.')
+else:
+    print('There is an error in the translation of functional.')
```

Comment:
Without reading all of it: I think this example goes far beyond the scope of the examples we have otherwise. I would prefer having the important parts of the code with some explanatory text in the online doc, and this example more concise and to the point.